base/io.jl

```diff
@@ -661,54 +661,89 @@ function readuntil(s::IO, delim::T; keep::Bool=false) where T
     return out
 end
 
-# requires that indices for target are small ordered integers bounded by start and endof
+# requires that indices for target are the integer unit range from start to endof
 # returns whether the delimiter was matched
-function readuntil_indexable(io::IO, target#=::Indexable{T}=#, out)
-    T = eltype(target)
-    first = start(target)
-    if done(target, first)
+# uses the Knuth–Morris–Pratt_algorithm, with the first and second cache entries unrolled
+# For longer targets, the cache improves the big-O efficiency of scanning of sequences
+# with repeated patterns
+# Each cache entry tells us which index we should start the search at.
+# We assume this is unlikely, so we only lazy-initialize as much of the cache as we need to use
+# When we allocate the cache, we initialize it to 0 (and offset by the first index afterwards).
+# Suppose target is:
+#    Index:  1245689
+#    Value: "aδcaδcx"
+# We would set the cache to
+#    0 0 0 1 0 3 4 0
```

Comment:
This is no longer accurate with your change since the `1 0 3` is `1 2 3` if you're looking for code units.